docs/docker_walkthrough.rst

```diff
@@ -0,0 +1,684 @@
+      --project current
+
+Then adapt the NLU start command in your docker compose so that it uses your
+NLU configuration. As in for the training mount the ``config`` directory into
+your NLU container and instruct Rasa NLU to use this configuration by adding
+the flag ``-c <path to your nlu config>``.
+The configuration of the ``rasa_nlu`` server might then look similar to this:
+
+.. code-block:: yaml
+
+  rasa_nlu:
+      image: rasa/rasa_nlu:latest-spacy
+      volumes:
+        - ./models/rasa_nlu:/app/models
+        - ./config:/app/config
+      command:
+        - start
+        - --path
+        - models
+        - -c
+        - config/nlu_config.yml
+
+.. note::
+
+  Depending on the selected pipeline you might have to use a different
+  Rasa NLU image.
+
+4. Adding Custom Actions
+------------------------
+
+To create more sophisticated chatbots you will probably use :ref:`customactions`.
+Continuing the example from above you might want to add an action which tells
+the user a joke to cheer the user up.
+
+4.1 Creating a Custom Action
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Start with creating the custom actions in a directory ``actions``.
+
+.. code-block:: bash
+
+  mkdir actions
+  # Rasa Core SDK expects a python module.
+  # Therefore, make sure that you have this file in the direcotry.
+  touch __init__.py
+  touch actions/actions.py
+
+Then build a custom action using the Rasa Core SDK, e.g.:
+
+.. code-block:: python
+
+  import requests
+  import json
+  from rasa_core_sdk import Action
+
+
+  class ActionJoke(Action):
+    def name(self):
+      return "action_joke"
+
+    def run(self, dispatcher, tracker, domain):
+      request = requests.get('http://api.icndb.com/jokes/random').json() #make an api call
+      joke = request['value']['joke'] #extract a joke from returned json response
+      dispatcher.utter_message(joke) #send the message back to the user
+      return []
+
+Then add the custom action in your stories and your domain file.
+Continuing the example from above replace replace ``utter_cheer_up`` in
+``data/stories.md`` with the custom action ``action_joke`` and add
+``action_joke`` to the actions in the domain file.
+
+4.2 Adding the Action Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The custom actions are run by the action server.
+To spin it up together with Rasa Core and Rasa NLU, add a service
+``action_server`` to the ``docker-compose.yml``.
+
+.. code-block:: yaml
+
+  action_server:
+    image: rasa/rasa_core_sdk:latest
+    volumes:
+      - ./actions:/app/actions
+
+This pulls the image for the Rasa Core SDK which includes the action server,
+mounts your custom actions into it, and starts the server.
+
+As for Rasa NLU, it is necessary to tell Rasa Core the location of the action
+server. Add this to your ``config/endpoints.yml``:
+
+.. code-block:: yaml
+
+  action_endpoint:
+    url: http://action_server:5055/webhook
+
+If you then run ``docker-compose up`` the action server starts together
```

Comment:
I would rephrase this as a mandatory step: 'Run 'docker-compose up' to start the action server together with...'.